wrappers/nodejs/src/addon.cpp

```diff
@@ -1544,7 +1586,7 @@ class RSSensor : public Nan::ObjectWrap {
 
     auto me = Nan::ObjectWrap::Unwrap<RSSensor>(instance);
     me->sensor_ = sensor;
-    me->options_ = new Options(reinterpret_cast<rs2_options*>(sensor));
+    me->SetOptionsPointer(reinterpret_cast<rs2_options*>(sensor));
```

Comment:
Any better way than Reinterpret cast?

Reply:
In fact, rs2_sensor inherits from rs2_options, so we need no cast in theory. 
But that inheriting relationship is declared in librealsense/src/rs.cpp.
The compiler only know these two types are of two unrelated types, thus causing static_cast and dynamic_cast all failed to compile. We could only use reinterpret_cast.

Reply:
is the rs2_sensor and rs2_options are multiple inherent?  If not, dynamic_cast should work.

Reply:
It's not multiple inheritance, the inheritance relationship was hidden from us:
1. the two types and the inheritance relationship are defined in librealsense/src/rs.cpp:
 ```
 struct rs2_options  {  ...  } 
 struct rs2_sensor : public rs2_options { ... }
```
2. the type declaration were in include/librealsense2/h/rs_types.h:
```
typedef struct rs2_sensor rs2_sensor; 
typedef struct rs2_options rs2_options; 

```

As we are only including the rs_types.h header file, the compiler only knows they are two distinct types, without knowing their inheritance relationship, so dynamic_cast will report error.

Reply:
@dorodnic Any thoughts on this?

Reply:
To not block the PR, I'd like to merge it and leave a TODO for further discussion.  @tingshao could you update your code with this TODO?

Reply:
@halton Ok.